examples/cpp/customserializable/main.cpp

```diff
@@ -0,0 +1,133 @@
+
+class Order : public PdxSerializable {
+ public:
+  Order() {
+    orderID = (unsigned long)0;
+    quantity = (unsigned int)0;
+    name = "";
+  };
+
+  Order(unsigned long inOrderID, const std::string& inName,
+        unsigned int inQuantity)
+      : orderID(inOrderID), name(inName), quantity(inQuantity){};
+
+  virtual ~Order(){};
+
+  unsigned long getOrderID() { return orderID; }
+  const std::string& getName() const { return name; }
+  unsigned int getQuantity() { return quantity; }
+
+  virtual void fromData(PdxReader& pdxReader) override {
+    try {
+      orderID = (unsigned long)pdxReader.readLong(orderid_key);
+      name = pdxReader.readString(name_key);
+      quantity = (unsigned int)pdxReader.readInt(quantity_key);
+    } catch (std::exception& e) {
+      std::cout << "Caught exception: " << e.what() << std::endl;
+      return;
+    }
+  }
+
+  virtual void toData(PdxWriter& pdxWriter) const override {
+    try {
+      pdxWriter.writeLong(orderid_key, orderID);
+      pdxWriter.markIdentityField(orderid_key);
+
+      pdxWriter.writeString(name_key, name);
+      pdxWriter.markIdentityField(name_key);
+
+      pdxWriter.writeInt(quantity_key, quantity);
+      pdxWriter.markIdentityField(quantity_key);
+
+    } catch (std::exception& e) {
+      std::cout << "Caught exception: " << e.what() << std::endl;
+      return;
+    }
+    return;
+  }
+
+  static PdxSerializable* createDeserializable() { return new Order(); }
+
+  virtual std::string toString() const override {
+    std::stringstream ss("");
+    ss << " OrderID: " << orderID << " Product Name: " << name
+       << " Quantity: " << quantity;
+    return ss.str();
+  }
+
+  virtual size_t objectSize() const override {
+    auto objectSize = sizeof(Order);
+    return objectSize;
+  }
+
+  const std::string& getClassName() const override {
+    static std::string className = "com.example.Order";
+    return className;
+  }
+
+ private:
+  unsigned long orderID;
+  std::string name;
+  unsigned int quantity;
+
+  const std::string orderid_key = "orderid";
+  const std::string name_key = "name";
+  const std::string quantity_key = "quantity";
+};
+
+int main(int argc, char** argv) {
+
+  auto cacheFactory = CacheFactory();
+  cacheFactory.set("log-level", "none");
+  auto cache = cacheFactory.create();
+
+  auto poolFactory = cache.getPoolManager().createFactory();
+  poolFactory->addLocator("localhost", 10334);
+  auto pool = poolFactory->create("pool");
+  auto regionFactory = cache.createRegionFactory(PROXY);
+  auto region = regionFactory.setPoolName("pool").create("custom_orders");
+
+  cache.getTypeRegistry().registerPdxType(Order::createDeserializable);
+
+  std::cout << "Create orders" << std::endl;
+  auto order1 = std::shared_ptr<Order>(new Order(1, "product x", 23));
+  auto order2 = std::shared_ptr<Order>(new Order(2, "product y", 37));
+
+  std::cout << "Storing orders in the region" << std::endl;
+  auto keyptr = CacheableKey::create("Customer1");
```

Comment:
You don't need to do this. Use the same example below using the raw string.

Reply:
done.